zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/UserTaskController.java

```diff
@@ -120,4 +130,34 @@ private static ProblemDetail mapRejectionToProblem(final BrokerRejection rejecti
         }
     };
   }
+
+  @PostMapping(
+      path = "/user-tasks/search",
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE},
+      consumes = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<UserTaskSearchQueryResponse> searchUserTasks(
+      @RequestBody(required = false) final UserTaskSearchQueryRequest query) {
+    return SearchQueryRequestMapper.toUserTaskQuery(query == null? new UserTaskSearchQueryRequest() : query)
+        .fold(
+            (Function<? super UserTaskQuery, ? extends ResponseEntity<UserTaskSearchQueryResponse>>)
```

Comment:
❓ Why is this necessary? Does the compiler fail otherwise? Just wondering as it looks a bit odd.

Reply:
The casting of the lambda in the fold method is necessary due to type inference limitations in Java generics, especially when dealing with Either types and lambdas. The fold method is designed to take two functions: one to handle the Left value and another to handle the Right value. Without the explicit cast, the compiler may not be able to infer the correct types for these functions.

However, the casting makes the code harder to read and understand. To avoid this, we can adjust the method signature to return ResponseEntity<Object>, which can encapsulate both UserTaskSearchQueryResponse and ProblemDetail.